spec/controllers/api/v1/aggregations_controller_spec.rb

```diff
@@ -1,169 +1,87 @@
-            get :index, params: { workflow_ids: ids.join(',') }
-            expect(result_workflow_ids).to match_array( [ "#{workflow.id}" ])
-          end
-        end
-
-        context "filtering on subject_id" do
-          let(:subject_id) { "#{aggregations.last.subject_id}" }
-
-          it "should return all the aggregated data for the supplied workflow" do
-            get :index, params: { workflow_id: workflow.id, subject_id: subject_id }
-            aggregate_failures "filtering" do
-              resources = json_response[api_resource_name]
-              expect(resources.length).to eq(1)
-              aggregation = resources.first
-              expect(aggregation["links"]["subject"]).to eq(subject_id)
-            end
-          end
-        end
-      end
-    end
+    it_behaves_like 'is indexable'
   end
 
-  describe "#show" do
+  describe '#show' do
     let(:resource) { create(:aggregation) }
     let(:authorized_user) { resource.workflow.project.owner }
 
-    it_behaves_like "is showable"
-
-    context "non-logged in users" do
-
-      context "when the workflow does not have public aggregation" do
-        let(:workflow) { create(:workflow) }
-
-        it "should return not_found" do
-          get :show, params: { id: resource.id }
-          expect(response).to have_http_status(:not_found)
-        end
-      end
-
-      context "when the workflow has the public aggregation flag" do
-
-        it "should return the aggregated resource with a public workflow" do
-          resource.workflow.update_column(:aggregation, { public: true })
-          get :show, params: { id: resource.id }
-          aggregate_failures "public show" do
-            expect(json_response[api_resource_name].length).to eq(1)
-            expect(created_instance(api_resource_name)["id"]).to eq("#{resource.id}")
-          end
-        end
-      end
-    end
+    it_behaves_like 'is showable'
   end
 
-  describe "create" do
-    let(:workflow) { create(:workflow_with_subjects) }
-    let(:subject) { workflow.subject_sets.first.subjects.first }
+  describe 'create' do
+    let(:workflow) { create(:workflow) }
     let(:authorized_user) { workflow.project.owner }
-    let(:test_attr) { :aggregation }
-    let(:test_attr_value) { { "something" => "HERE",
-                              "workflow_version" => "1.1" } }
+    let(:test_attr) { :workflow_id }
+    let(:test_attr_value) { workflow.id }
+    let(:fake_response) { double }
+    let(:mock_agg) { instance_double(AggregationClient) }
 
     let(:create_params) do
       { aggregations:
           {
-            aggregation: test_attr_value,
             links: {
-              subject: subject.id.to_s,
+              user: authorized_user.id.to_s,
               workflow: workflow.id.to_s
             }
           }
       }
     end
 
-    it_behaves_like "is creatable"
+    before do
+      allow(AggregationClient).to receive(:new).and_return(mock_agg)
```

Comment:
Not blocking, but maybe we could add a test where calling out with AggregationClient errors that ensures we receive the correct service unavailable response?

Reply:
✔️